File: 401_examples/struct_malloc_example/CICCCStudent.c
#include <stdlib.h>
#include "CICCCStudent.h"

// Implementation file

CICCCStudent* CICCCStudent_new(int _age, char* _name, char _grade)
{
  CICCCStudent* this = (CICCCStudent*) malloc(sizeof(CICCCStudent));
  (*this).age = _age;
  this->name = _name;
  this->grade = _grade;

  return this;
}

void CICCCStudent_release(CICCCStudent* this)
{
  free(this);
}

File: 401_objective-c_examples/ObjectiveCExample02/ObjectiveCExample02/CICCCStudent.m
//
//  CICCCStudent.m
//  ObjectiveCExamble02
//
//  Created by Luiz Fernando Peres on 2016-01-12.
//  Copyright © 2016 Ideia do Luiz. All rights reserved.
//

#import "CICCCStudent.h"

@interface CICCCStudent ()

@end

@implementation CICCCStudent

- (instancetype)initStudentWithAge:(NSInteger)age
{
    self = [super init];
    
    if (self)
    {
        _age = age;
    }
    
    return self;
}

- (instancetype)initStudentWithAge:(NSInteger)theAge name:(NSString *)theName andGrade:(char)theGrade
{
    self = [super init];
    
    if (self)
    {
        _age = theAge;
        _name = theName;
        _grade = theGrade;
    }
    
    return self;
}

- (NSString *)description
{
    return [NSString stringWithFormat:@"The student name is %@, the age is %lu and the grade is %c", self.name, self.age, self.grade];
}

@end